tests/user/test_user_auth.py

```diff
@@ -0,0 +1,151 @@
+import base64
```

Comment:
Tests are great here :)

I'd add testes for the new types of exceptions I suggested in the main code.